sdk/search/azure-search/src/samples/java/com/azure/search/DataSourceExample.java

```diff
@@ -0,0 +1,157 @@
+
+    public static void main(String[] args) {
+        SearchServiceClient client = new SearchServiceClientBuilder()
+            .endpoint(ENDPOINT)
+            .credential(new SearchApiKeyCredential(ADMIN_KEY))
+            .buildClient();
+
+        /*
+         * Store the names of the created data sources so that we can delete them later
+         * without affecting other resources.
+         * */
+        Collection<String> names = new HashSet<>();
+
+        names.add(createSqlDataSource(client));
+        names.add(createCosmosDataSource(client));
+        names.add(createBlobDataSource(client));
+        names.add(createTableStorageDataSource(client));
+
+        /*
+         * Get all existing data sources; list should include the ones we just created.
+         * */
+        PagedIterable<DataSource> dataSources = client.listDataSources();
+        for (DataSource dataSource : dataSources) {
+            if (names.contains(dataSource.getName())) {
+                System.out.println(String.format("Found data source %s of type %s", dataSource.getName(), dataSource.getType().toString()));
+            }
+        }
+
+        /*
+         * Delete the data sources we just created.
+         * */
+        for (String name : names) {
+            deleteDataSource(client, name);
+        }
+    }
+
+    private static void deleteDataSource(SearchServiceClient client, String dataSourceName) {
+        try {
+            client.deleteDataSource(dataSourceName);
+        } catch (Exception ex) {
+            System.err.println(ex.toString());
+        }
+    }
+
+    private static DataSource createSampleDatasource(DataSourceType type,
+                                                     String connectionString,
+                                                     DataContainer container,
+                                                     DataChangeDetectionPolicy dataChangeDetectionPolicy) {
+        return new DataSource()
+            .setName(generateDataSourceName())
+            .setType(type)
+            .setCredentials(new DataSourceCredentials()
+                .setConnectionString(connectionString))
+            .setContainer(container)
+            .setDataChangeDetectionPolicy(dataChangeDetectionPolicy);
+    }
+
+    private static String createDataSource(
+        SearchServiceClient client,
+        DataSourceType type,
+        String connectionString,
+        DataContainer container,
+        DataChangeDetectionPolicy dataChangeDetectionPolicy) {
+
+        DataSource dataSource = createSampleDatasource(type, connectionString, container, dataChangeDetectionPolicy);
+        try {
+            client.createOrUpdateDataSource(dataSource);
+        } catch (Exception ex) {
+            System.err.println(ex.toString());
+        }
+        return dataSource.getName();
+    }
+
+    private static String createTableStorageDataSource(SearchServiceClient client) {
```

Comment:
Any reason we didn't use the helper functions that were provided for each of these data source types?

Reply:
In this sepcific sample the purpuse is to demonstrate how the user can utlize the model classes. More samples can be added (backlog updated)